src/hunspell/CMakeLists.txt

```diff
@@ -0,0 +1,172 @@
+
+
+# ---------------------------------------------------------------------------- #
+# ~~~ Shared library target
+add_library(${HUNSPELL_LIB_TARGET_SHARED} SHARED)
+
+target_link_libraries(${HUNSPELL_LIB_TARGET_SHARED}
+    PRIVATE
+        ${HUNSPELL_LIB_TARGET_OBJ}
+        # Cmake passes linker flags through compiler
+        $<$<CXX_COMPILER_ID:GNU>:-Wl,--no-undefined>
+        #   $<$<CXX_COMPILER_ID:Clang>:-Wl,-no-undefined,error>
+)
+target_include_directories(${HUNSPELL_LIB_TARGET_SHARED}
+    INTERFACE  ${HUNSPELL_LIB_INCLUDE_DIRECTORIES}
+)
+
+# Properties
+set_target_properties(${HUNSPELL_LIB_TARGET_SHARED} PROPERTIES
+    OUTPUT_NAME  ${HUNSPELL_LIB_OUTPUT_NAME}
+    VERSION      0.0.${PROJECT_VERSION_PATCH}  # ${PROJECT_VERSION}
+    SOVERSION    ${PROJECT_VERSION_MAJOR}
+    FOLDER       ${HUNSPELL_LIB_TARGET_PREFIX}
+)
+
+# Create alias
+add_library(${HUNSPELL_NAMESPACE}${HUNSPELL_LIB_TARGET_SHARED} ALIAS ${HUNSPELL_LIB_TARGET_SHARED})
+
+# Add install target
+add_install_target(${HUNSPELL_LIB_TARGET_SHARED})
+# ---------------------------------------------------------------------------- #
+
+
+# ---------------------------------------------------------------------------- #
+# ~~~ Static library target
+option(HUNSPELL_LIB_STATIC_PIC     "Build static library with position independent code."  ON)
+option(HUNSPELL_LIB_STATIC_HIDDEN  "Set the visibility of all symbols to hidden."          OFF)
+mark_as_advanced(
+    HUNSPELL_LIB_STATIC_PIC
+    HUNSPELL_LIB_STATIC_HIDDEN
+)
+
+if(HUNSPELL_BUILD_STATIC_LIB)
+    add_library(${HUNSPELL_LIB_TARGET_STATIC} STATIC)
+    
+    # Archive the already created object files
+    target_link_libraries(${HUNSPELL_LIB_TARGET_STATIC}
+        PRIVATE  ${HUNSPELL_LIB_TARGET_OBJ}
+    )
+    target_include_directories(${HUNSPELL_LIB_TARGET_STATIC}
+        INTERFACE  ${HUNSPELL_LIB_INCLUDE_DIRECTORIES}
+    )
+    target_compile_definitions(${HUNSPELL_LIB_TARGET_STATIC}
+        INTERFACE  "HUNSPELL_STATIC"
+    )
+
+    # ~~~ Recompile source files if flags set
+    if((NOT HUNSPELL_LIB_STATIC_PIC) OR HUNSPELL_LIB_STATIC_HIDDEN)
+        set_target_properties(${HUNSPELL_LIB_TARGET_STATIC} PROPERTIES
+            SOURCES  "${HUNSPELL_LIB_SOURCES}"
+        )
+    endif()
+
+    if(NOT HUNSPELL_LIB_STATIC_PIC)
+        # Force target PIC flag to OFF in case CMAKE_POSITION_INDEPENDENT_CODE is set
+        set_target_properties(${HUNSPELL_LIB_TARGET_STATIC} PROPERTIES
+            POSITION_INDEPENDENT_CODE  OFF
+        )
+    endif()
+
+    if(HUNSPELL_LIB_STATIC_HIDDEN)
+        # Set the visibility of *all* FUNC symbols
+        set_target_properties(${HUNSPELL_LIB_TARGET_STATIC} PROPERTIES
+            CXX_VISIBILITY_PRESET  hidden
+        )
+    endif()
+
+    # ~~~ Output name and version
+    set_target_properties(${HUNSPELL_LIB_TARGET_STATIC} PROPERTIES
+        OUTPUT_NAME  ${HUNSPELL_LIB_OUTPUT_NAME}$<$<BOOL:${WIN32}>:_static>
```

Comment:
Please do not duplicate CMake built-in functionality: https://cmake.org/cmake/help/latest/variable/CMAKE_STATIC_LIBRARY_SUFFIX.html

The user will set this if they need it, but the idiomatic naming scheme on *nix is `lib<name>.a` and `<name>.lib` on Windows.